### PR TITLE
Optimization

### PR DESCRIPTION
**Pull request by tsvietOK**
Removed code duplication
Reworked selection system

- Previous selection method worked the following way: user selects item and then file has been reselected programmatically. This patch fixes that. 
- Use of `FileList.SelectedItem` replaced with `SelectedItems` as we use "Extended" selection mode (look at microsoft [docs](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.gridview?view=winrt-19041#remarks))
- `BaseLayout.SelectedItem` is just first element of `BaseLayout.SelectedItems`
- Moved some functions to UI abstract methods
- Removed code duplication
- Removed cast to `BaseLayout` as `ContentPage` is typeof `BaseLayout`

### CODE REVIEW

**Review by lukeblevins — APPROVED** [review]
I like these changes. Maybe when you get a chance, you can take a look at the "optimizations" branch that exists already (incomplete and quite a bit behind master) and see if there's any work in there that you want to pick up. So far, the focus was making better use of a partial class BaseLayout to contain the interactions. Totally up to you, though.

**Comment by tsvietOK** [comment]
@duke7553 I think `BaseLayout` should contain only UI-related code. `Intercation` should contain functions like copy, paste, pin item and other.
We have to decide what we will do with this PR, because if you merge this, you (or someone else) will have to port this changes to `optimization` branch to get merge back to master without conflicts.

**Comment by yaichenbaum** [comment]
@tsvietOK Would you like to join the UWP Discord so we can collaborate on Files UWP outside of issues and pull requests? https://discord.gg/Km25JR

**Comment by tsvietOK** [comment]
@yaichenbaum Joined

**Comment by yaichenbaum** [comment]
@duke7553 Do we want to merge this pull request or should we wait?

**Comment by lukeblevins** [comment]
@yair100  I like these changes a lot. But we should wait until #827 is merged to prevent conflicts with his work.

File: Files/Strings/en-US/Resources.resw
Comment (yaichenbaum) [inline Files/Strings/en-US/Resources.resw]: Can you revert the changes to these strings?

Comment (tsvietOK) [reply Files/Strings/en-US/Resources.resw]: Fixed by 301387b76bc15c19b091335177b1225e179fcbc8

**Review by yair100 — APPROVED** [review]

**Pull request merged**
